processor/src/operations/sys_ops.rs

```diff
@@ -122,9 +131,62 @@ impl Process {
     {
         self.stack.copy_state(0);
         self.decoder.set_user_op_helpers(Operation::Emit(event_id), &[event_id.into()]);
-        host.on_event(self.into(), event_id)?;
 
-        Ok(())
+        // If it's a native event, handle it directly. Otherwise, forward it to the host.
+        if let Some(advice_injector) = AdviceInjector::from_event_id(event_id) {
+            self.handle_advice_injector(advice_injector, host)
+        } else {
+            host.on_event(self.into(), event_id)
+        }
```

Comment:
Not for this PR, but in the future we should probably split `event_id` into source and ID where source identifies a library (e.g., `source=0` for the system events, `source=1` for the standard library etc.). We've discussed some of this in https://github.com/0xPolygonMiden/miden-vm/issues/1457, but may be good to create a separate issue once #1457 is closed.

Reply:
My issue with this though is that it reduces the number of "non-colliding libraries" from potentially 2^32 down to 2^16. Also, most libraries will probably only define a few events, wasting ~2^16 bits of unused event IDs (but still allocated to them).

I figured it to be better to just have libraries subscribe to the set of event IDs they need, (which the host can track with a `BTreeMap<u32, EventHandler>`). Then, we're actually very unlikely for libraries to define the same event ID (assuming all libraries properly generate their event IDs randomly).

Reply:
Let's discuss this in a follow-up issue. For example, we could still have 32-bit space for libraries and then within each library we could have a 16-bit space for event IDs.